packages/gui/src/components/walletConnect/WalletConnectAddConnectionDialog.tsx

```diff
@@ -107,6 +139,70 @@ export default function WalletConnectAddConnectionDialog(props: WalletConnectAdd
   const isStepValid = step === Step.CONNECT || selectedFingerprints.length > 0;
   const canSubmit = !isSubmitting && !isLoading && isStepValid;
 
+  function renderKeysMultiSelect() {
+    return (
+      <Select
+        multiple
+        value={selectedFingerprints}
+        sx={{
+          '.MuiSelect-select .Mui-checked': {
+            display: 'none',
+          },
+        }}
+        native={false}
+        renderValue={() => t`Select keys`}
```

Comment:
We should limit the height. I have 30 or so keys, causing the menu listing to take up the entire height of the screen.

```suggestion
        renderValue={() => t`Select keys`}
        MenuProps={{
          PaperProps: {
            style: {
              maxHeight: '250px',
              overflowY: 'auto',
            },
          },
        }}
```